WordPress/gutenberg
# DataViews: Allow actions to be provided declaratively as a prop

Related #55083 

## What?

This is a small refactoring/API PR that extracts the "actions" property of the data views into a dedicated prop instead of leaving it as part of "fields".

The reasoning is simple:

 - Down the road we want to be able to reuse actions as "bulk actions".
 - We want third-party developers to be able to register actions.
 - Fields should be usable in other layouts (like Grid) where actions are rendered completely differently, so there's no point in merging "fields" with "actions".

## Testing Instructions

- Nothing should change here, the "trash post" action should continue to work as expected.
- you can try passing an empty array as "actions" and see that the actions column disappears entirely.
